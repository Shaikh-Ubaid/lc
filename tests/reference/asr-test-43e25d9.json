{
    "basename": "asr-test-43e25d9",
    "cmd": "lc --show-asr --no-color {infile} -o {outfile} -extra-arg=\"\"",
    "infile": "tests/test.cpp",
    "infile_hash": "9e5ef4c37b7e9e1f13f0082ce65459c1e596430887537eb8fd656b77",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-test-43e25d9.stdout",
    "stdout_hash": "f31a3fab00e8917eaae30983c0ed97a62e13a54b839a6439e6e7b225",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}