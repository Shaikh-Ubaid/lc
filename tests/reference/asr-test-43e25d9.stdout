(TranslationUnit
    (SymbolTable
        1
        {
            f:
                (Function
                    (SymbolTable
                        2
                        {
                            __return_var:
                                (Variable
                                    2
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            result:
                                (Variable
                                    2
                                    result
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            x:
                                (Variable
                                    2
                                    x
                                    []
                                    InOut
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    f
                    (FunctionType
                        [(Integer 4)]
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    [(Var 2 x)]
                    [(=
                        (Var 2 result)
                        (IntegerBinOp
                            (Var 2 x)
                            Div
                            (IntegerConstant 42 (Integer 4))
                            (Integer 4)
                            ()
                        )
                        ()
                    )
                    (=
                        (Var 2 __return_var)
                        (Var 2 result)
                        ()
                    )
                    (Return)]
                    (Var 2 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                )
        })
    []
)
